pmd-core/src/main/java/net/sourceforge/pmd/Report.java

```diff
@@ -436,6 +426,10 @@ public boolean hasMetrics() {
         return metrics.iterator();
     }
 
+    /**
```

Comment:
Let's spend a sentence to explain, what this method checks:

`Checks whether there are no violations and no processing errors.
That means, that PMD analysis yielded nothing to worry about.`